### PR TITLE
Editor enhancements

### LINKED ISSUE #9071: semi-colon in sql queries causes error in the ui
When executing SQL queries which have semi-colons causes an error. The Editor was very permissive about executing multiple SQL queries with semi-colons, like multiple update statements.

screen shot 2016-07-21 at 1 58 11 pm

### PR DESCRIPTION
**Pull request by xavijam**
- [x] Don't apply empty content from the editor (CartoCSS or SQL), requested from https://github.com/CartoDB/cartodb/pull/9262#issuecomment-237203212.
- [x] Don't apply SQL with ; at the end (fixes #9071).

CR: @nobuti

### CODE REVIEW

**Comment by xavijam** [comment]
ping! :)

File: lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js [hunk 1]
```diff
@@ -184,8 +184,21 @@ module.exports = CoreView.extend({
...
+    var appliedQuery = this._querySchemaModel.get('query');
+    var currentQuery = this._codemirrorModel.get('content');
+
+    // Remove last character if it has ';'
+    if (currentQuery && currentQuery.slice(-1) === ';') {
+      currentQuery = currentQuery.slice(0, currentQuery.length - 1);
+      this._codemirrorModel.set('content', currentQuery);
+    }
+
+    var isSameQuery = currentQuery.toLowerCase() === appliedQuery.toLowerCase();
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js]: We can we apply the helper you made yesterday, right?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/dataset/dataset-options/dataset-options-view.js]: I had my reticences, but I don't remember them today, so yeah, I'll apply it here, thanks.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/data/data-view.js [hunk 2]
```diff
@@ -128,10 +128,18 @@ module.exports = CoreView.extend({
...
+      currentQuery = currentQuery.slice(0, currentQuery.length - 1);
+      this._codemirrorModel.set('content', currentQuery);
+    }
+
+    var isSameQuery = currentQuery.toLowerCase() === appliedQuery.toLowerCase();
     var altersData = SQLUtils.altersData(currentQuery);
     var notification;
 
-    if (currentQuery.toLowerCase() === appliedQuery.toLowerCase()) {
+    if (!currentQuery || isSameQuery) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/data/data-view.js]: IMO falsy comparison doesn't read well here, I'd make `currentQuery === ''`. When I see `!currentQuery` I automatically start thinking in `undefined`, but in this case, if undefined, isSameQuery would had failed.

**Comment by nobuti** [comment]
👍

**Comment by xavijam** [comment]
Feedback applied :)

File: lib/assets/javascripts/cartodb3/helpers/sql-utils.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+
 /**
  *  Checks properties from a SQL
  */
 
 module.exports = {
-
   isSameQuery: function (originalQuery, customQuery) {
-    if (!originalQuery || !customQuery) {
+    if (originalQuery === null || originalQuery === undefined || customQuery === null || customQuery === undefined) {
```
Comment (nobuti) [inline lib/assets/javascripts/cartodb3/helpers/sql-utils.js]: `originalQuery == null` already checks `null` and `undefined`

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/helpers/sql-utils.js]: ❤️

**Pull request merged**